crates/sui-node/src/lib.rs

```diff
@@ -1,11 +1,15 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use anemo_tower::callback::CallbackLayer;
+use anemo_tower::trace::TraceLayer;
 use anyhow::anyhow;
 use anyhow::bail;
 use anyhow::Result;
 use futures::TryFutureExt;
 use mysten_network::server::ServerBuilder;
+use narwhal_network::metrics::MetricsMakeCallbackHandler;
```

Comment:
Do we want shared networking code to live outside narwhal somewhere?

Reply:
Yes i think we probably do, see my comment [here](https://github.com/MystenLabs/sui/pull/5277#discussion_r997274598) for how we can proceed with this.